src/utils.py

```diff
@@ -0,0 +1,153 @@
+
+    Returns:
+        go.Figure: A Plotly figure object.
+    
+    """
+
+    df = pd.DataFrame({
+        'Feature': data.X.columns,
+        'Feature Relevance': data.get_relevance(),
+    })
+
+    # Calculate the statistics if showing redundancy
+    display_text = []
+    hover_cols = {'Feature Relevance': False}
+
+    # When displaying a solution, show selected feature as solid and non-selected
+    # features as transparent
+    if selected_features:
+        opacity = np.repeat(0.3, len(df))
+    else:
+        opacity = np.repeat(1.0, len(df))
+        selected_features = []
+    opacity[selected_features] = 1.0
+
+    mlw = np.repeat(1 if data.n < 50 else 0, len(df))
+    if data.n < 50:
+        mlw[selected_features] = 3
+
+    # Manually calculate the continuous color map to show redundancy.
+    # This is required for different opacity levels per bar.
+    if hover_data and show_red:
+        idx = hover_data['points'][0]['pointIndex']
+        redundancy_data = data.get_redundancy()
+        # Protect against case where the last hovered point was from a larger data set.
+        if idx < data.n:
+            # Note: this modifies global data and so is not compatible with use
+            # of the demo in a multi-user environment.  One alternative is to
+            # make a copy of the DataFrame prior to modifying the redundancy
+            # column.
+            df['Redundancy'] = redundancy_data[idx]
+            hover_cols['Redundancy'] = False
+
+        color_data = df['Redundancy'].values
+        normalized_color_data = (color_data - np.min(color_data)) / (np.max(color_data) - np.min(color_data))
+        rgba_colors = sample_colorscale(COLOR_SCALE, normalized_color_data, colortype='rgb')
+
+        for i in range(len(rgba_colors)):
+            rgba_colors[i] = 'rgba' + rgba_colors[i][3:-1] + ', ' + str(opacity[i]) + ')'
+
+        if data.n < 30:
+            display_text = [round(i.item(),2) for i in color_data]
+    else:
+        rgba_colors = [f'rgba(42, 125, 225, {o})' for o in opacity]
+        
+    # Plot the bar graph
+    fig = go.Figure(data=[go.Bar(
+        x=df["Feature"], 
+        y=df["Feature Relevance"], 
+        text=display_text,
+        textposition='outside',
+    )])
+
+    fig.update_traces(
+        marker_color=rgba_colors,
+        marker_line_color='black',
+        marker_line_width=mlw,
+        hoverinfo='none', 
+        hovertemplate=None,
+    )
+    
+    fig.update_layout(
+        font=dict(size=GRAPH_FONT_SIZE),
+        yaxis_title='Feature Relevance to Outcome',
+        yaxis_range=[0,1.1],
+    )
+    
+    # Modify axis labels:
+    if data.name == 'titanic':
+        fig.update_layout(xaxis_title='Passenger Features')
+    elif data.name == 'scene':
+        fig.update_layout(xaxis_title='Color and Texture Features in Image')
+
+    return fig
+
+def draw_accuracy_bars(data: DataSet, selected_features: list, soln_score: float) -> go.Figure:
```

Comment:
Again only add this if it can be None 

```suggestion
def draw_accuracy_bars(data: DataSet, selected_features: Optional[list], soln_score: float) -> go.Figure:
```